source/common/http/http3/conn_pool.cc

```diff
@@ -76,11 +76,11 @@ Http3ConnPoolImpl::Http3ConnPoolImpl(
     const Network::TransportSocketOptionsConstSharedPtr& transport_socket_options,
     Random::RandomGenerator& random_generator, Upstream::ClusterConnectivityState& state,
     CreateClientFn client_fn, CreateCodecFn codec_fn, std::vector<Http::Protocol> protocol,
-    OptRef<PoolConnectResultCallback> connect_callback)
+    OptRef<PoolConnectResultCallback> connect_callback, Http::PersistentQuicInfo& quic_info)
     : FixedHttpConnPoolImpl(host, priority, dispatcher, options, transport_socket_options,
                             random_generator, state, client_fn, codec_fn, protocol),
-      quic_info_(Quic::createPersistentQuicInfoForCluster(dispatcher, host->cluster())),
-      server_id_(getConfig(host->transportSocketFactory()).serverNameIndication(),
+      quic_info_(dynamic_cast<Quic::PersistentQuicInfoImpl&>(quic_info)),
```

Comment:
I'm slightly surprised we need to cast from the interface to the impl here. Should we perhaps widen the interface so that it supports everything that this class needs from it?

Reply:
Oh, I see. Ugh! Man, I sure hate ENVOY_ENABLE_QUIC. Fair enough, though.